### PR TITLE
GH-37466: [C++][Parquet] Fix Valgrind failure in DELTA_BYTE_ARRAY decoder

### LINKED ISSUE #37466: [C++][Parquet] Valgrind error in TestDeltaByteArrayEncodingDirectPut
### Describe the bug, including details regarding any error messages, version, and platform.

See failed nightly build at https://github.com/ursacomputing/crossbow/runs/16329060808

### Component(s)

C++, Continuous Integration, Parquet

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

Avoid skipping any padding bits where reading a one-value data page, as it doesn't contain any block (the value is encoded in the header).

### Are these changes tested?

Yes, by existing tests and CI jobs.

### Are there any user-facing changes?

No.

* Closes: #37466

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit -g cpp

File: cpp/src/parquet/encoding.cc [hunk 1]
```diff
@@ -2522,32 +2522,37 @@ class DeltaBitPackDecoder : public DecoderImpl, virtual public TypedDecoder<DTyp
...
+        total_values_remaining_ -= max_values;
+        this->num_values_ -= max_values;
+        return max_values;
+      }
+      InitBlock();
+    }
+
+    DCHECK(first_block_initialized_);
     while (i < max_values) {
+      // Ensure we have an initialized mini-block
```
Comment (mapleFU) [inline cpp/src/parquet/encoding.cc:2549]: comment should before 2547?

By the way, is this related to the valgrind error?

File: cpp/src/arrow/util/bit_stream_utils.h [hunk 2]
```diff
@@ -108,34 +108,39 @@ class BitWriter {
   int bit_offset_;   // Offset in buffered_values_
 };
 
+namespace detail {
+
+inline uint64_t ReadLittleEndianWord(const uint8_t* buffer, int bytes_remaining) {
```
Comment (wgtmac) [inline cpp/src/arrow/util/bit_stream_utils.h:113]: Will this have issue if `bytes_remaining` is 0?

Comment (pitrou) [reply cpp/src/arrow/util/bit_stream_utils.h:113]: Probably not, as long as `buffer` is non-null.
Note that this is merely a small refactor of `BitReader` but may not be necessary to fix this issue.

**Review by wgtmac — APPROVED** [review]

**Comment by pitrou** [comment]
CI failures are unrelated, I'll merge.

**Pull request merged**